### PR TITLE
Remove unneeded inheritance in InstallHelper.cs

### LINKED ISSUE #767: InstallHelper should not inherit from PSCmdlet
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://www.powershellgallery.com/packages/PowerShellGet)
- [X] Search the existing issues.

### Steps to reproduce

In `InstallHelper.cs` on line 30, it inherits from `PSCmdlet`. This is incorrect because it's a class that contain functionality, not the cmdlet itself. 
It should also not throw terminating errors or write errors.

### Expected behavior

```console
InstallHelper doesn't inherit from PSCmdlet and doesn't throw terminating errors or write errors from that class.
```

### Actual behavior

```console
InstallHelper inherits from PSCmdlet and throws terminating errors or write errors from that class.
```

### Error details

_No response_

### Environment data

```powershell
ModuleType Version    PreRelease Name                                PSEdition ExportedCommands
---------- -------    ---------- ----                                --------- ----------------
Script     2.2.5                 PowerShellGet                       Desk      {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability…}

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.19044
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}       
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by alyssa1303**
# PR Summary

This PR removes inheritance of the class `InstallHelper` from `PSCmdlet`. This PR also fixes tests in `InstallPSResource`, `SavePSResource`, and `UpdatePSResource` to make sure the errors thrown are matched. 

## PR Context

Resolves #767 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed]()
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [ ] N/A or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary.

### CODE REVIEW

File: test/InstallPSResource.Tests.ps1 [hunk 1]
```diff
@@ -461,14 +461,25 @@ Describe 'Test Install-PSResource for Module' {
     # Install module that is not authenticode signed
     # Should FAIL to install the  module
     It "Install module that is not authenticode signed" -Skip:(!(Get-IsWindows)) {
-        Install-PSResource -Name $testModuleName -Version "5.0.0" -AuthenticodeCheck -Repository $PSGalleryName -TrustRepository -ErrorAction SilentlyContinue
-        $Error[0].FullyQualifiedErrorId | Should -be "InstallPackageFailed,Microsoft.PowerShell.PowerShellGet.Cmdlets.InstallPSResource"
+        try 
+        {
+            Install-PSResource -Name $testModuleName -Version "5.0.0" -AuthenticodeCheck -Repository $PSGalleryName -TrustRepository -ErrorAction SilentlyContinue
```
Comment (PaulHigin) [inline test/InstallPSResource.Tests.ps1]: We can remove the try/catch and use Pester `Should -Throw -ErrorId` pattern.
```suggestion
            { Install-PSResource -Name $testModuleName -Version "5.0.0" -AuthenticodeCheck -Repository $PSGalleryName -TrustRepository } | Should -Throw -ErrorId 'GetAuthenticodeSignatureError,Microsoft.PowerShell.PowerShellGet.Cmdlets.InstallPSResource'

```

**Review by PaulHigin — CHANGES_REQUESTED** [review]

File: test/InstallPSResource.Tests.ps1 (hunk 1 above)
Comment (alyssa1303) [reply test/InstallPSResource.Tests.ps1]: I see. I'll fix that for all other tests

Comment (PaulHigin) [reply test/InstallPSResource.Tests.ps1]: Don't forget to remove the outer `try/catch` block.  Pester takes care of catching the thrown exception and comparing the `fullqualifiederrorid`, and so the try/catch block is not needed.

**Review by PaulHigin — APPROVED** [review]
LGTM

**Review by alerickson — APPROVED** [review]

**Pull request merged**